useScroll/HoverToScroll.js

```diff
@@ -0,0 +1,196 @@
+	} : {
+		axis: 'x',
+		client: 'clientWidth',
+		maxPosition: 'maxLeft',
+		scrollPosition: 'scrollLeft'
+	};
+};
+const hoverToScrollMultiplier = 0.04;
+const directionToFocus = {
+	horizontal: {
+		before: 'right',
+		after: 'left'
+	},
+	vertical: {
+		before: 'down',
+		after: 'up'
+	}
+};
+
+/**
+ * A hover area to scroll for a single direction
+ *
+ * @class HoverToScrollBase
+ * @memberof sandstone/useScroll/HoverToScroll
+ * @ui
+ * @private
+ */
+const HoverToScrollBase = (props) => {
+	const {direction, scrollContainerHandle} = props;
+	const bounds = scrollContainerHandle.current.getScrollBounds();
+
+	// Mutable value
+
+	const mutableRef = useRef({
+		hoverToScrollRafId: null,
+		hoveredPosition: null
+	});
+
+	// Functions
+
+	const handleGlobalKeyDown = useCallback((ev) => {
+		const keyCode = ev.keyCode;
+		let position = mutableRef.current.hoveredPosition;
+
+		if (scrollContainerHandle.current.rtl && direction === 'horizontal') {
+			position = position === 'after' ? 'before' : 'after';
+		}
+
+		if (getDirection(keyCode) || is('enter', keyCode)) {
+			Spotlight.focusNextFromPoint(
+				directionToFocus[direction][position],
+				getLastPointerPosition()
+			);
+			scrollContainerHandle.current.stop();
+		}
+	}, [direction, scrollContainerHandle]);
+
+	const startRaf = useCallback((job) => {
+		if (typeof window === 'object') {
+			mutableRef.current.hoverToScrollRafId = window.requestAnimationFrame(job);
+			if (typeof document === 'object') {
+				document.addEventListener('keydown', handleGlobalKeyDown, {capture: true});
+			}
+		}
+	}, [handleGlobalKeyDown]);
+
+	const stopRaf = useCallback(() => {
+		if (typeof window === 'object' && mutableRef.current.hoverToScrollRafId !== null) {
+			window.cancelAnimationFrame(mutableRef.current.hoverToScrollRafId);
+			mutableRef.current.hoverToScrollRafId = null;
+			if (typeof document === 'object') {
+				document.removeEventListener('keydown', handleGlobalKeyDown, {capture: true});
+			}
+		}
+	}, [handleGlobalKeyDown]);
+
+	const scrollJob = useCallback((position) => {
+		if (typeof window === 'object') {
+			const {axis, client, maxPosition, scrollPosition} = propertyNames(direction);
+			const distance =
+				(position === 'before' ? -1 : 1) * // scroll direction
+				bounds[client] * // scroll page size
+				hoverToScrollMultiplier; // a scrolling speed factor
+
+			return function (ev) {
+				if (ev.pointerType === 'mouse') {
+					mutableRef.current.hoveredPosition = position;
+					const scrollByHover = () => {
+						if (getLastInputType() === 'mouse') {
+							scrollContainerHandle.current.scrollTo({
+								position: {
+									[axis]: clamp(
+										0,
+										bounds[maxPosition],
+										scrollContainerHandle.current[scrollPosition] + distance
+									)
+								},
+								animate: false
+							});
+							startRaf(scrollByHover);
+						} else {
+							stopRaf(); // for other type input during hovering
+						}
+					};
+					startRaf(scrollByHover);
```

Comment:
I didn't find any bugs about the behavior, but is it intended that startRaf() is called twice?